unidist/core/backends/mpi/core/controller/api.py

```diff
@@ -174,18 +178,29 @@ def init():
                 # https://www.intel.com/content/www/us/en/docs/mpi-library/developer-reference-linux/2021-8/other-environment-variables.html.
                 os.environ["I_MPI_SPAWN"] = "1"
 
-            if hosts:
+            host_list = hosts.split(",") if hosts is not None else ["localhost"]
+            host_count = len(host_list)
+
+            if common.is_shared_memory_supported():
+                # +1 for monitor process on each host
+                nprocs_to_spawn = cpu_count + len(host_list)
+            else:
+                # +1 for just a single process monitor
+                nprocs_to_spawn = cpu_count + 1
+            if host_count > 1:
```

Comment:
We should not allow single host MpiHosts because:
1. If the value is the current host - this parameter is useless;
2. If the value is another host, we will not be able to launch it correctly, because Root process is already running.

This validation has already added.

Reply:
Pleaes create an issue where we dicuss the option of usage for passing a single host to MpiHosts.

Reply:
the issue has been created with #337